test/HelmetDeclarativeTest.js

```diff
@@ -2501,6 +2501,41 @@ describe("Helmet - Declarative API", () => {
         });
     });
 
+    describe("deferred tags", () => {
+        beforeEach(() => {
+            window.__spy__ = sinon.spy();
+        });
+
+        afterEach(() => {
+            delete window.__spy__;
+        });
+
+        it("executes synchronously when defer={true} and async otherwise", done => {
+            ReactDOM.render(
+                <Helmet>
+                    <script defer={false}>
```

Comment:
Good point. Maybe it should just go to `<Helmet>`, then the user can decide whether to separate tags into the `defer` `<Helmet>` or not. Would make the code easier to support as well.